apps/prairielearn/assets/scripts/instructorFileEditorClient.ts

```diff
@@ -98,6 +111,92 @@ class InstructorFileEditor {
         .querySelector<HTMLButtonElement>('.js-reformat-file')
         ?.addEventListener('click', () => this.reformatJSONFile());
     }
+
+    // Override the save button click to show confirmation modal if needed
+    this.saveElement!.addEventListener('click', (e) => this.handleSaveClick(e));
+  }
+
+  /**
+   * Handles the save button click, showing a confirmation modal if there are issues,
+   * or proceeding with the save if the user has already confirmed.
+   */
+  handleSaveClick(event: MouseEvent) {
+    if (this.confirmedSave) {
+      this.confirmedSave = false;
+      return;
+    }
+
+    const issues = this.checkForSaveIssues();
+
+    if (issues.length > 0) {
+      event.preventDefault();
+      this.showConfirmationModal(issues);
+    }
+
+    // Otherwise, continue with the save
+  }
+
+  /**
+   * Checks for issues that would require confirmation before saving.
+   *
+   * @returns An array of issues that would require confirmation before saving.
+   */
+  checkForSaveIssues(): string[] {
+    const issues: string[] = [];
+    const currentContents = this.editor.getValue();
+
+    if (this.fileMetadata && this.fileMetadata.type !== FileType.File) {
+      try {
+        const parsedContent = JSON.parse(currentContents);
+
+        if (typeof parsedContent !== 'object') {
+          issues.push(
+            `The ${friendlyNameForFileType(this.fileMetadata.type).toLowerCase()} metadata cannot be parsed as JSON.`,
+          );
+        } else if (this.fileMetadata.uuid) {
+          if ('uuid' in parsedContent) {
+            if (parsedContent.uuid !== this.fileMetadata.uuid) {
+              issues.push(
+                `The UUID in this ${friendlyNameForFileType(this.fileMetadata.type).toLowerCase()} file should match "${this.fileMetadata.uuid}".`,
+              );
```

Comment:
Discussed on Slack, let's consider wording more like the following:

> The UUID was changed. This may result in data loss, as the current assessment will be deleted and a new one will be created with the new UUID.

We could consider saying "The UUID was changed from <tt>{old}</tt> to <tt>{new}</tt>", but that might be unnecessary.

Reply:
Outdated.